services/alert/service.go

```diff
@@ -769,7 +770,15 @@ func (s *Service) createHandlerFromSpec(spec HandlerSpec) (handler, error) {
 		}
 		h = NewPublishHandler(c, s.logger)
 	case "sensu":
-		h = s.SensuService.Handler(s.logger)
+		c := sensu.HandlerConfig{}
+		err = decodeOptions(spec.Options, &c)
+		if err != nil {
+			return handler{}, err
+		}
+		h, err := s.SensuService.Handler(c, s.logger)
```

Comment:
I found the bug it is super subtle, the above line should read

```
h, err = s.SensuService.Handler(c, s.logger)
```

this is because we want to assign to the var h Handler from the function scope not our local scope. I should redesign this so that isn't so easy to mess up, but for now it will fix your test.